### PR TITLE
Return early if all images are already present on all nodes

### PR DESCRIPTION
**Pull request by Alfa-Alex**
Solves #2413.

### CODE REVIEW

**Comment by Alfa-Alex** [comment]
I signed it

**Comment by Alfa-Alex** [comment]
I signed it.

File: pkg/cmd/kind/load/docker-image/docker-image.go [hunk 1]
```diff
@@ -141,6 +141,11 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 		}
 	}
 
+	// return early if no node needs the image
```
Comment (BenTheElder) [inline pkg/cmd/kind/load/docker-image/docker-image.go:144]: unrelated to this PR but the code above also looks suspect ... where we check `len(selectedNodes)` when logging about a particular image even though nodes may have been selected for another image, will address in a follow-up.

Comment (BenTheElder) [reply pkg/cmd/kind/load/docker-image/docker-image.go:144]: selectedNodes also really ought to be a set, clearly should have reviewed this code more carefully.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
thank you!

**Comment by BenTheElder** [comment]
rootless docker at least in github actions is currently known broken https://github.com/kubernetes-sigs/kind/issues/2409

**Pull request merged**